pkg/driver/node/credentialprovider/provider_pod.go

```diff
@@ -227,13 +228,18 @@ func (c *Provider) createIRSACredentialsEnvironment(ctx context.Context, provide
 }
 
 // createEKSPodIdentityCredentialsEnvironment creates an environment with the environment variables needed for pod-level authentication with EKS Pod Identity
-func (c *Provider) createEKSPodIdentityCredentialsEnvironment(provideCtx ProvideContext) envprovider.Environment {
+func (c *Provider) createEKSPodIdentityCredentialsEnvironment(provideCtx ProvideContext) (envprovider.Environment, error) {
 	podID := provideCtx.GetCredentialPodID()
 	tokenName := podLevelEksPodIdentityServiceAccountTokenName(podID, provideCtx.VolumeID)
 	tokenFile := filepath.Join(provideCtx.EnvPath, tokenName)
 
+	eksPodIdentityAgentCredentialsURI := os.Getenv("EKS_POD_IDENTITY_AGENT_CONTAINER_CREDENTIALS_FULL_URI")
+	if eksPodIdentityAgentCredentialsURI == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "Failed to detect EKS_POD_IDENTITY_AGENT_CONTAINER_CREDENTIALS_FULL_URI driver configuration flag")
```

Comment:
Maybe we can emit an error log saying "seems like EKS PI is disabled, if you want to use it please provide `eksPodIdentityAgent.containerCredentialsFullURI` Helm value"